### PR TITLE
First version

### PR DESCRIPTION
**Pull request by gabmontes**
Add the first version of the code to the repo, along with tests and the standard checks like linter, code formatter, etc.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  ],
+  "main": "src/index.js",
+  "repository": "hemilabs/esplora-api-client",
+  "scripts": {
+    "coverage": "c8 npm test",
+    "format:check": "prettier --check .",
+    "knip": "knip",
+    "lint": "eslint --cache .",
+    "prepare": "husky",
+    "prepublishOnly": "npm run build",
```
Comment (gndelia) [inline package.json]: `build` command is missing in the list. (By the way, is there anything to build? The repo is plain js)

File: package.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "ignoreDependencies": [
+      "eslint-config-prettier",
+      "lint-staged"
+    ]
+  },
+  "lint-staged": {
+    "*.{js,md}": "eslint --cache --fix",
+    "*.{js,json,md}": "prettier --write"
+  },
+  "prettier": {}
```
Comment (gndelia) [inline package.json:136]: is this needed for the project to pick up prettier's defaults? Or can it be removed?

File: types/index.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ */
+
+/**
+ * @typedef {object} EsploraClient
+ * @property {object} bitcoin
+ * @property {AddressesMethods} bitcoin.addresses
+ * @property {BlocksMethods} bitcoin.blocks
+ * @property {FeesMethods} bitcoin.fees
+ * @property {TransactionsMethods} bitcoin.transactions
+ */
```
Comment (gndelia) [inline types/index.js:98]: Contrarian 😅

File: package.json (hunk 2 above)
Comment (gabmontes) [reply package.json:136]: Needed so VSCode users can keep the Prettier plugin "Require Config" set and auto-format the code.

File: package.json (hunk 1 above)
Comment (gabmontes) [reply package.json]: Nothing to build 😄 

Removed!

**Review by gndelia — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+## Installation
+
+```sh
+npm install esplora-client
+```
+
+## Usage
+
+```js
+import { esploraApiClient } from "esplora-client";
```
Comment (gndelia) [inline README.md]: I think now this is:

```suggestion
import { esploraClient } from "esplora-client";
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+const { bitcoin } = esploraApiClient({ network: "testnet" });
```
Comment (gndelia) [inline README.md]:
```suggestion
const { bitcoin } = esploraClient({ network: "testnet" });
```

**Review by gndelia — CHANGES_REQUESTED** [review]

**Review by gndelia — APPROVED** [review]

**Review by mbonfoster — APPROVED** [review]

**Pull request merged**